numpy/lib/npyio.py

```diff
@@ -1006,8 +1006,8 @@ def read_data(chunk_size):
             if len(vals) == 0:
                 continue
             if usecols:
-                vals = [vals[j] for j in usecols]
-            if len(vals) != ncols:
+                vals = usecols_getter(vals)
+            elif len(vals) != ncols:
```

Comment:
Can you explain this change to an `elif`?

Reply:
If usecols is not None, then we already know that usecols_getter will (by construction) return a list with the right number of elements.  The performance gain is negligible, but it seems just good to skip the unneeded check.